federator/src/lib/Heartbeat.js

```diff
@@ -66,6 +62,132 @@ module.exports = class Heartbeat {
         }
     }
 
+    async readLogs() {
+        await this._checkIfRsk()
+        let retries = 3;
+        const sleepAfterRetrie = 3000;
+        while(retries > 0) {
+            try {
+                const currentBlock = await this.mainWeb3.eth.getBlockNumber();
+                const fedContract = await this.federationFactory.getMainFederationContract();
+
+                const toBlock = currentBlock;
+                this.logger.info('Running to Block', toBlock);
+
+                if (toBlock <= 0) {
+                    return false;
+                }
+
+                if (!fs.existsSync(this.config.storagePath)) {
+                    fs.mkdirSync(this.config.storagePath);
+                }
+                let originalFromBlock = this.config.mainchain.fromBlock || 0;
+                let fromBlock = null;
+                try {
+                    fromBlock = fs.readFileSync(this.lastBlockPath, 'utf8');
+                } catch(err) {
+                    fromBlock = originalFromBlock;
+                }
+                if(fromBlock < originalFromBlock) {
+                    fromBlock = originalFromBlock;
+                }
+                if(fromBlock >= toBlock){
+                    this.logger.warn(`Current chain Height ${toBlock} is the same or lesser than the last block processed ${fromBlock}`);
+                    return false;
+                }
+                fromBlock = parseInt(fromBlock)+1;
+                this.logger.debug('Running from Block', fromBlock);
+
+                const recordsPerPage = 1000;
+                const numberOfPages = Math.ceil((toBlock - fromBlock) / recordsPerPage);
+                this.logger.debug(`Total pages ${numberOfPages}, blocks per page ${recordsPerPage}`);
+
+                var fromPageBlock = fromBlock;
+                for(var currentPage = 1; currentPage <= numberOfPages; currentPage++) {
+                    var toPagedBlock = fromPageBlock + recordsPerPage-1;
+                    if(currentPage == numberOfPages) {
+                        toPagedBlock = toBlock
+                    }
+
+                    this.logger.debug(`Page ${currentPage} getting events from block ${fromPageBlock} to ${toPagedBlock}`);
+                    const heartbeatLogs = await fedContract.getPastEvents('HeartBeat', {
+                        fromBlock: fromPageBlock,
+                        toBlock: toPagedBlock
+                    });
+
+                    if (!heartbeatLogs) throw new Error('Failed to obtain HeartBeat logs');
+                    await this._processHeartbeatLogs(
+                        heartbeatLogs,
+                        {
+                            ethLastBlock: currentBlock
```

Comment:
should use ethereum blocknumber (in this case await this.mainWeb3.eth.getBlockNumber();)